newIDE/app/src/VariablesList/index.js

```diff
@@ -256,24 +274,44 @@ export default class VariablesList extends React.Component<Props, State> {
           if (!parentVariable) {
             variablesContainer.remove(name);
           } else {
-            parentVariable.removeChild(name);
+            if (type === gd.Variable.Structure)
+              parentVariable.removeChild(name);
+            else parentVariable.removeAtIndex(index);
           }
 
           this.forceUpdate();
           if (this.props.onSizeUpdated) this.props.onSizeUpdated();
         }}
-        onAddChild={() => {
-          const name = newNameGenerator('ChildVariable', name =>
-            variable.hasChild(name)
-          );
-          variable.getChild(name).setString('');
+        onAddChild={(forceType: ?string) => {
+          if (typeof forceType === 'string') {
+            if (forceType === 'structure') {
+              const name = newNameGenerator('ChildVariable', name =>
+                variable.hasChild(name)
+              );
+              variable.getChild(name).setString('');
+            } else
+              variable.getAtIndex(variable.getChildrenCount()).setString('');
```

Comment:
I wonder if we should have a `push` method in `gd::Variable`? 
So that it's clear what is done, and to encourage using `getAtIndex` with an index that is checked to be inside the bounds (in most languages you'll get an exception/crash/error if you try to access an out of bound. While we don't want a crash in GDevelop, we should still access only in the bounds, and use a separate function to add an item to the array. This can help optimization).

Reply:
Yes, it sounds like a good idea. It is also necessary if we want to resurrect GDCpp and implement the new types in it.